fstapi/src/reader.rs

```diff
@@ -283,7 +283,7 @@ pub enum Hier<'a> {
 impl<'a> Hier<'a> {
   /// Creates a new hierarchy.
   fn new(hier: &'a capi::fstHier) -> Self {
-    match hier.htyp as u32 {
+    match hier.htyp as c_int  {
```

Comment:
This will fail on Linux, please check the CI output.

I think the following modification is okay:


```suggestion
    match hier.htyp as capi::fstHierType  {
```